### PR TITLE
Add ThreadPool.parallel_map; add tests for parallel and parallel_map

### PR DESCRIPTION
**Pull request by twrodriguez**

### CODE REVIEW

File: lib/opengov/util/thread_pool.rb [hunk 1]
```diff
@@ -55,4 +38,37 @@ def join
     @pool.each(&:join)
     @pool.clear
   end
+
+  private
```
Comment (jpterry) [inline lib/opengov/util/thread_pool.rb]: This private declaration ins't actually doing anything.

If you defined all these class methods in `class << self` you could take advantage of `private` the right way and maybe you wouldn't need to name this method `_parallel_exec` to indicate its privateness.

File: lib/opengov/util/thread_pool.rb [hunk 2]
```diff
@@ -55,4 +38,37 @@ def join
...
+          end
+        rescue Timeout::Error => e
+          thread_returns[return_key] = e
+        end
+      end
+    end
+    pool.join
+    thread_returns
+  end
+  private_class_method :_parallel_exec
```
Comment (jpterry) [inline lib/opengov/util/thread_pool.rb]: oh. I see. See above :)

File: lib/opengov/util/thread_pool.rb [hunk 3]
```diff
@@ -6,30 +6,13 @@ class << self
   # Default Value
   @concurrency_limit = Float::INFINITY.to_f
```
Comment (jpterry) [inline lib/opengov/util/thread_pool.rb]: Maybe we could have a sane default on this? On OS X this default is 2046, on linux it is a function of physical memory,  245355 on a worker in current prod. Both of those values are pretty crazy for what I'd expect as default behavior of a thread pool. Just a thought. 

`cat /proc/sys/kernel/threads-max` How about that

File: lib/opengov/util/thread_pool.rb [hunk 4]
```diff
@@ -55,4 +32,50 @@ def join
...
+          return_key = index
+        end
+
+        pool.push do
+          begin
+            Timeout.timeout(opts[:timeout]) do
+              thread_returns[return_key] = yield(item)
+            end
+          rescue Timeout::Error => e
+            thread_returns[return_key] = e
```
Comment (jpterry) [inline lib/opengov/util/thread_pool.rb]: It might be nice to have the option to raise this instead. Possibly even in the wrapping methods?

**Comment by cademarkegard** [comment]
lgtm

**Pull request merged**